src/Sylius/Bundle/ApiBundle/StateProvider/AdminResetPasswordProvider.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\StateProvider;
+
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProviderInterface;
+use Sylius\Bundle\CoreBundle\Message\Admin\Account\ResetPassword;
+
+/** @experimental */
+final class AdminResetPasswordProvider implements ProviderInterface
+{
+    public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null
+    {
+        if (!is_a($operation->getClass(), ResetPassword::class, true)) {
+            return null;
+        }
+
```

Comment:
```suggestion
```
I would be in favor of making `Processors/Providers` less "defensively programmed".
In APIP2 we had a `supports` method because APIP iterated over all the services. Now we bind these services explicitly, so I don't know if this checking is really necessary (it can only protect the developer from misconfiguration and will never affect the application itself).

WDYT? @Sylius/team